domains/DataCollection/Forms/PatientID/index.js

```diff
@@ -23,10 +27,52 @@ import configArray from './config';
 // });
 
 const PatientIDForm = ({ navigation }) => {
+  // similar to componentDidMount and componenetWillUnmount
+  // runs every 10 seconds in the background to get all Async Data
+  useEffect(() => {
+    const interval = setInterval(() => {
```

Comment:
this whole chain of logic needs to be put into a `util.js` file probably. it's a tad long. make a `utils` folder in the domain `PatientID` and export a function that handles all this logic so we can remove it from the form

separation of concerns